src/main.ts

```diff
@@ -50,10 +51,12 @@ async function main() {
       core.warning(
         "public_github_token not set, GitHub API requests may be limited",
```

Comment:
I'm wondering if we should move this warning to only show if we are actually about to make an API request (either in `latestVersion` or `downloadBuf`)? Otherwise we might emit the warning in cases where it doesn't matter (e.g. the user has installed buf some other way, or it is already in the tool cache).

Reply:
Moved to `downloadBuf` as that is the only case where a rate limit may apply .

Reply:
It seems like rate limits are larger for REST api requests too if you are authed: https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users

Reply:
On construction of the public REST client we must provide an auth token (based on the user error). So only downloads currently. We may be able to relax this but would need testing. Resolving for now.